tests/unit/plugins/test_go.py

```diff
@@ -47,6 +49,21 @@ def setUp(self):
             target_deb_arch=self.deb_arch, snapcraft_yaml_file_path=snapcraft_yaml_path
         )
 
+        self.mocked_temp_dir = "tempdir"
+
+        @contextlib.contextmanager
+        def tempdir():
+            self.tempdir = self.mocked_temp_dir
+            os.mkdir(self.tempdir)
+            # fake the creation of the "vet" binary used in tests.
+            open(os.path.join(self.tempdir, "vet"), "w").close()
+
+            yield self.tempdir
+
+        self.useFixture(
+            fixtures.MockPatch("tempfile.TemporaryDirectory", side_effect=tempdir)
+        )
```

Comment:
Use of tempdir is now gone, this has been further simplified, but no, it would not have been sufficient, we needed to mock the is in the go plugin and add a side effect on instantiation of the context used there